File: CivitasR/lib/civitas/casilla_sorpresa.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative "casilla.rb"
require_relative "sorpresa_ir_carcel.rb"
require_relative "sorpresa_ir_casilla.rb"
require_relative "sorpresa_pagar_cobrar.rb"
require_relative "sorpresa_por_casa_hotel.rb"
require_relative "sorpresa_por_jugador.rb"
require_relative "sorpresa_salir_carcel.rb"
require_relative "sorpresa.rb"
require_relative "jugador.rb"
#require "byebug"
module Civitas
  class CasillaSorpresa < Casilla
    
    
#    /**
#     * @brief constructor de casilla sorpresa
#     * @param mazo de cartas sorpresa
#     * @param nombre
#     */ 
    def initialize (mazo, nombre)
      super(nombre)
      @mazo = mazo
    end
    
    
    def recibeJugador( actual, todos)
      #byebug
        if(jugadorCorrecto(actual, todos))
            sorpresa = @mazo.siguiente
            informe(actual, todos)
            sorpresa.aplicarAJugador(actual,todos)
        end
    end
    
    #    /**
#     * @brief represente con detalle la información acerca de la casilla
#     */
    def toString()
      text = super
      if @sorpresa !=nil
       text = text + "\nLa Sorpresa Contiene: " + @sorpresa.toString
      end
      return text;
    end   
    
  end
end

File: CivitasJ/src/civitas/CasillaSorpresa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author ELVIRA
 */
public class CasillaSorpresa extends Casilla{
       
    
    /**
     * @brief constructor de casilla sorpresa
     * @param mazo de cartas sorpresa
     * @param nombre
     */ 
    CasillaSorpresa (MazoSorpresas mazo, String nombre){
        super(nombre);
        super.setMazo (mazo);
    }
    
    
    
    @Override
    void recibeJugador(int actual, ArrayList<Jugador>todos){
        if(jugadorCorrecto(actual, todos)){
            System.out.println("Has caido en una casilla sorpresa!!");
            setSorpresa (getMazo().siguiente());
            informe(actual, todos);
            getSorpresa().aplicarAJugador(actual,todos);
        }
    }
    
 /**
     * @brief represente con detalle la información acerca de la casilla
     */
    @Override
    public String toString(){
        String text = super.toString() +
        "\n LA SORPRESA CONTIENE: " + getSorpresa().toString();
        //if(mazo != null) //mazo no tiene tostring
        //    text = text + "\n mazo: " + mazo.toString();       
        return text;
    }       
    
}
